cheatsheets/Microservices_based_Security_Arch_Doc_Cheat_Sheet.md

```diff
@@ -0,0 +1,148 @@
+| Access type | Specify access type, e.g. “Read” or “Read/Write”
+
+### Identify “service-to-service” synchronous communications
+
+Collect information on the parameters listed below related to each “service-to-service” synchronous communication.
+
+| Parameter name | Description | 
+| :--- | :--- |
+| Caller service name (ID) | Specify caller service name (ID) defined above
+| Called service name (ID) | Specify called service name (ID) defined above
+| Protocol/framework used| Specify protocol/framework used for communication, e.g. HTTP (REST, SOAP), Apache Thrif, gRPC
+| Short description | Shortly describe the purpose of communication (requests for query of information or request/commands for a state-changing business function) and data passed between services (if possible, in therms of assets defined above)
+
+### Identify “service-to-service” asynchronous communications
+
+Collect information on the parameters listed below related to each “service-to-service” asynchronous communication.
+
+| Parameter name | Description | 
+| :--- | :--- |
+| Publisher service name (ID) | Specify publisher service name (ID) defined above
+| Subscriber service name (ID) | Specify subscriber service name (ID) defined above
+| Message queue (ID) | Specify message queue (ID) defined above
+| Short description | Shortly describe the purpose of communication (receiving of information or commands for a state-changing business function) and data passed between services (if possible, in therms of assets defined above)
+
+### Identify “asset-to-storage” relations
+
+Collect information on the parameters listed below related to each “asset-to-storage” relation.
+
+| Parameter name | Description | 
+| :--- | :--- |
+| Asset name (ID) | Asset name (ID) defined above
+| Storage name (ID) | Specify storage name (ID) defined above
+| Storage type | Specify storage type for the asset, e.g. "golden source" or "cache"
+
+## Create a graphical presentation of application architecture
+
+It is advisable to create graphical presentation of application architecture (building blocks and relations defined above) in form of services call graph or data flow diagram. In order to do that one can use special software tools (e.g. Enterprise Architect) or [DOT language](https://en.wikipedia.org/wiki/DOT_(graph_description_language)). See example of using DOT language [here](https://gist.github.com/vladgolubev/80c5523336ddec3859c0e90d9a070882).
+
+## Use collected information in secure software development practices
+
+Collected information may be useful for doing application security practices, e.g. during defining security requirements, threat modeling or security testing. Table below contains examples of activities related to securing application architecture (as well as its mapping to OWASP ASVS 4.0) and tips for their implementation using information collected above.
+
+| Activity description | Implementation tips | Mapping to OWASP ASVS 4.0 |

```

Comment:
I am wondering if the table is the best option here. For me 'Implementation tips' are poorly readable. 
One way to improve readability is to use bullet points inside table e.g.

| Activity description | Implementation tips | Mapping to OWASP ASVS 4.0 |
| :--- | :--- | :--- |
| Attack surface analysis | To enumerate microservices endpoints that need to be tested during security testing and analyzed during threat modeling analyze data collected under the following sections: <ul><li>"Identify and describe application-functionality services" (parameter “API definition”) </li><li>"Identify and describe infrastructure services" (parameter “Link to the service documentation”)</li></ul> | 1.1.2

But I do not know how to do this without HTML syntax inside table...

Another approach will be to refactor it to other form than table.
What do you think @righettod @AlexB1986 @ThunderSon

Reply:
Thanks for your time and review!

Yes, my first version (Word) was with that bullet points.
We may refactor it in the following way:


## Use collected information in secure software development practices

....
### Attack surface analysis
#### Implementation tips
To enumerate microservices endpoints that need to be tested during security testing and analyzed during threat modeling analyze data collected under the following sections:
* "Identify and describe application-functionality services" (parameter “API definition”);
* "Identify and describe infrastructure services" (parameter “Link to the service documentation”).
#### Mapping to OWASP projects
* [OWASP ASVS, V1 “Architecture, Design and Threat Modeling Requirements”, #1.1.2](https://github.com/OWASP/ASVS/blob/master/4.0/en/0x10-V1-Architecture.md#v1-architecture-design-and-threat-modeling-requirements)
* [OWASP Attack Surface Analysis Cheat Sheet](https://github.com/OWASP/CheatSheetSeries/blob/master/cheatsheets/Attack_Surface_Analysis_Cheat_Sheet.md)

Reply:
I think like you guys, table is not well suitable for this kind of formatting of content.
The refactoring proposed by @AlexB1986 is better.
We only use HTML in very rare condition when there is no other way to achieve an expected result.

Reply:
@AlexB1986 I also like the formatting that you proposed :)

Reply:
OK, will update it soon

Reply:
Done